### PR TITLE
Define message header

### PR DESCRIPTION
**Pull request by naturetime10**

### CODE REVIEW

**Comment by calper-ql** [comment]
Naming seems to be a little odd.

**Comment by calper-ql** [comment]
Also socket listen is blocking. Once you set all the callbacks you want to create a thread and detach listener.

File: protocol/message-header.proto [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        DATA = 1;
+    }
+
+    enum Service {
+        COMPUTER_VISION = 0;
+        MACHINE_LEARNING = 1;
+        GEOMETRY_PROCESSING = 2;
+        RENDERING = 3;
+        HEAD_TRACKING = 4;
+        IMAGE_PROVIDING = 5;
```
Comment (binamkayastha) [inline protocol/message-header.proto:18]: What's image_providing again? Is that directions images from the phone or something?

Comment (binamkayastha) [reply protocol/message-header.proto:18]: Thanks for merging without answering my question!

Comment (naturetime10) [reply protocol/message-header.proto:18]: This refers to Kinect server

**Pull request merged**